src/utils.cc

```diff
@@ -400,9 +400,16 @@ bool utils::checkModel(const RooStats::ModelConfig &model, bool throwOnFail) {
         if (a->isConstant() || allowedToFloat.contains(*a)) continue;
         if (a->getAttribute("flatParam")) {
             RooRealVar *rrv = dynamic_cast<RooRealVar *>(a);
-            if (rrv->getVal() > rrv->getMax() || rrv->getVal() < rrv->getMin()) {
-                ok = false; errors << "ERROR: flatParam " << rrv->GetName() << " has a value " << rrv->getVal() << 
-                                      " outside of the defined range [" << rrv->getMin() << ", " << rrv->getMax() << "]\n";
+            RooCategory *cat = dynamic_cast<RooCategory *>(a);
```

Comment:
I'm probably missing something, but why do we need this? Why would we declare a `RooCategory` parameter as a `flatParam`?